src/storage/kv/store.rs

```diff
@@ -97,30 +146,32 @@ impl Store {
         Ok(())
     }
 
-    /// Closes the store. It sends a stop signal to the writer and waits for the done signal.
+    /// Closes the inner store
     pub async fn close(&self) -> Result<()> {
-        if self.is_closed.load(std::sync::atomic::Ordering::SeqCst) {
-            return Ok(());
+        if let Some(inner) = self.inner.as_ref() {
+            inner.close().await?;
         }
 
-        // Send stop signal
-        self.stop_tx
-            .send(())
-            .await
-            .map_err(|e| Error::SendError(format!("{}", e)))?;
+        Ok(())
+    }
+}
 
-        // Wait for task to finish
-        if let Some(handle) = self.task_runner_handle.lock().await.take() {
-            handle.await.map_err(|e| {
-                Error::ReceiveError(format!(
-                    "Error occurred while closing the kv store. JoinError: {}",
-                    e
-                ))
-            })?;
-        }
+impl Drop for Store {
+    fn drop(&mut self) {
+        if !self._dropped {
+            // Replace the current self with a default, so we can send the original self to the async task
+            let mut this = Store::default();
+            std::mem::swap(&mut this, self);
+            this._dropped = true;
 
-        self.core.close()?;
-        Ok(())
+            // Close the store asynchronously
+            tokio::spawn(async move {
+                if let Err(err) = this.close().await {
```

Comment:
Im wondering if it does actually make sense to have this. I know the bug is that there is termination before a proper shutdown. But surely we should be building to tolerate such failures, instead of building to hope that this async drop is successful?